OpenRA.Mods.Common/Widgets/ViewportControllerWidget.cs

```diff
@@ -305,6 +310,66 @@ public override bool HandleKeyPress(KeyInput e)
 				return true;
 			}
 
+			if (key == ks.ViewPortBookmarkSaveSlot1)
+			{
+				viewPortBookmarkSlot1 = worldRenderer.Viewport.CenterPosition;
+				return false;
+			}
+
+			if (key == ks.ViewPortBookmarkSaveSlot2)
+			{
+				viewPortBookmarkSlot2 = worldRenderer.Viewport.CenterPosition;
+				return false;
+			}
+
+			if (key == ks.ViewPortBookmarkSaveSlot3)
+			{
+				viewPortBookmarkSlot3 = worldRenderer.Viewport.CenterPosition;
+				return false;
+			}
+
+			if (key == ks.ViewPortBookmarkSaveSlot4)
+			{
+				viewPortBookmarkSlot4 = worldRenderer.Viewport.CenterPosition;
+				return false;
+			}
+
+			if (key == ks.ViewPortBookmarkUseSlot1)
+			{
+				if (viewPortBookmarkSlot1.X != 0)
```

Comment:
@viszkoktamas keep in mind it seams you mean ```null``` (HEX: 0x00)  instead of ```0``` (HEX: 0x30)
the different is the first "0" is nothing and the second one is a printable char which is used to terminate strings (for example) :)